DEC/tests/DEC.jl

```diff
@@ -129,12 +57,36 @@ function save_dynamics(save_file_name)
   ax = CairoMakie.Axis(f[1,1], title = @lift("Heat at time $($time)"))
   gmsh = mesh!(ax, rect, color=h, colormap=:jet,
                colorrange=extrema(soln(tₑ)))
-  #Colorbar(f[1,2], gmsh, limits=extrema(soln(tₑ).h))
   Colorbar(f[1,2], gmsh)
   timestamps = range(0, tₑ, step=5.0)
   record(f, save_file_name, timestamps; framerate = 15) do t
     time[] = t
   end
 end
 
-end
\ No newline at end of file
+## 1-D HEAT EQUATION WITH DIFFUSIVITY
+
+function new_heat_equation(roe)
+  u = fresh!(roe, PrimalForm(0), :u)
+  k = fresh!(roe, Scalar(), :k) 
+
+  ∂ₜ(u) ≐ k * Δ(u)
+  
+  ([u], [k])
+end
+
+# we can reuse the mesh and operator lookup
+_vf = vfield(new_heat_equation, operator_lookup)
```

Comment:
this one won't run for me
```
_vf = vfield(new_heat_equation, operator_lookup) # XXX: brooken
ERROR: KeyError: key Scalar() * PrimalForm(0) not found
Stacktrace:
  [1] getindex
    @ ./dict.jl:477 [inlined]
  [2] (::DEC.Models.ThDEC.var"#toexpr#15"{Dict{TypedApplication, Any}, Dict{RootVar, Union{Expr, Symbol}}})(expr::SSAExpr)
    @ DEC.Models.ThDEC ~/.julia/dev/GATlab/DEC/src/models/ThDEC/ThDEC.jl:97
  [3] (::DEC.Models.ThDEC.var"#7#17")(::Any)
    @ DEC.Models.ThDEC ~/.julia/dev/GATlab/DEC/src/models/ThDEC/ThDEC.jl:126
  [4] iterate
    @ ./generator.jl:48 [inlined]
  [5] collect_to!
    @ ./array.jl:838 [inlined]
  [6] collect_to_with_first!
    @ ./array.jl:816 [inlined]
  [7] collect(itr::Base.Generator{Base.Iterators.Enumerate{Vector{SSAExpr}}, DEC.Models.ThDEC.var"#7#17"})
    @ Base ./array.jl:790
  [8] map
    @ ./abstractarray.jl:3402 [inlined]
  [9] vfield(model::Any, operator_lookup::Dict{TypedApplication, Any})
    @ DEC.Models.ThDEC ~/.julia/dev/GATlab/DEC/src/models/ThDEC/ThDEC.jl:125
 [10] top-level scope
    @ REPL[26]:1
```

Reply:
Right, @olynch and I reviewed this today and decided we need to just tweak the code to use Julia's * operation. This and the other comments should be resolved tomorrow